code/game/machinery/cryopod.dm

```diff
@@ -41,121 +41,117 @@
-	if(stat & (NOPOWER|BROKEN))
-		return
+/obj/machinery/computer/cryopod/attack_ghost(mob/user)
+	ui_interact(user)
 
-	user.set_machine(src)
-	add_fingerprint(usr)
+/obj/machinery/computer/cryopod/attack_hand(mob/user)
+	if(..())
+		return TRUE
+	add_fingerprint(user)
+	ui_interact(user)
 
-	var/dat
+/obj/machinery/computer/cryopod/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "CryopodConsole", name, 400, 480)
+		ui.open()
 
-	if(!( SSticker ))
-		return
+/obj/machinery/computer/cryopod/ui_data(mob/user)
+	var/list/data = list()
+	data["allow_items"] = allow_items
+	data["frozen_crew"] = frozen_crew
+	data["frozen_items"] = list()
 
-	dat += "<hr/><br/><b>[storage_name]</b><br/>"
-	dat += "<i>Welcome, [user.real_name].</i><br/><br/><hr/>"
-	dat += "<a href='?src=[UID()];log=1'>View storage log</a>.<br>"
 	if(allow_items)
-		dat += "<a href='?src=[UID()];view=1'>View objects</a>.<br>"
-		dat += "<a href='?src=[UID()];item=1'>Recover object</a>.<br>"
-		dat += "<a href='?src=[UID()];allitems=1'>Recover all objects</a>.<br>"
-
-	user << browse(dat, "window=cryopod_console")
-	onclose(user, "cryopod_console")
-
-/obj/machinery/computer/cryopod/Topic(href, href_list)
-	if(..())
-		return 1
+		data["frozen_items"] = frozen_items
 
-	var/mob/user = usr
 
-	add_fingerprint(user)
+	if(!ishuman(user))
+		return data
 
-	if(href_list["log"])
+	var/obj/item/card/id/id_card
+	var/mob/living/carbon/human/person = user
 
-		var/dat = "<b>Recently stored [storage_type]</b><br/><hr/><br/>"
-		for(var/person in frozen_crew)
-			dat += "[person]<br/>"
-		dat += "<hr/>"
+	id_card = person.get_idcard()
+	if(id_card?.registered_name)
+		data["account_name"] = id_card.registered_name
 
-		user << browse(dat, "window=cryolog")
+	return data
 
-	if(href_list["view"])
-		if(!allow_items) return
+/obj/machinery/computer/cryopod/ui_act(action, params, datum/tgui/ui)
+	if(..())
+		return
 
-		var/dat = "<b>Recently stored objects</b><br/><hr/><br/>"
-		for(var/obj/item/I in frozen_items)
-			dat += "[I.name]<br/>"
-		dat += "<hr/>"
+	if(stat & (NOPOWER|BROKEN))
+		return
 
-		user << browse(dat, "window=cryoitems")
+	var/mob/user = ui.user
 
-	else if(href_list["item"])
-		if(!allowed(user))
-			to_chat(user, "<span class='warning'>Access Denied.</span>")
-			return
-		if(!allow_items) return
+	add_fingerprint(user)
 
-		if(frozen_items.len == 0)
-			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
-			return
+	switch(action)
+		if("one_item")
+			if(!allowed(user))
+				to_chat(user, "<span class='warning'>Access Denied.</span>")
+				return
 
-		var/obj/item/I = input(usr, "Please choose which object to retrieve.","Object recovery",null) as null|anything in frozen_items
-		if(!I)
-			return
+			if(!allow_items)
+				return
```

Comment:
Can you move the
```dm
if(!allowed(user))
	to_chat(user, "<span class='warning'>Access Denied.</span>")
	return
	
if(!allow_items)
	return
```
to before the switch so we don't have to do it for each switch if statement?